File: selenium-framework/pages/cart_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage


class CartPage(BasePage):
    ROWS         = (By.CSS_SELECTOR, "#cart table tbody tr, .table-bordered tbody tr")
    REMOVE_BTN   = (By.CSS_SELECTOR, "button[aria-label='Remove'], .fa-times")
    UPDATE_BTN   = (By.CSS_SELECTOR, "button.btn-update, [data-bs-original-title='Update']")
    EMPTY_MSG    = (By.XPATH, "//*[contains(., 'cart is empty')]")
    CHECKOUT_BTN = (By.LINK_TEXT, "Checkout")

    def go(self):
        return self.open("checkout/cart&language=en-gb")

    def is_empty(self) -> bool:
        return self.is_visible(self.EMPTY_MSG)

    def line_count(self) -> int:
        if self.is_present(self.ROWS):
            return len(self.find_all(self.ROWS))
        return 0

    def proceed_to_checkout(self):
        self.click(self.CHECKOUT_BTN)
        return self

File: cypress-framework/cypress/pages/CartPage.js
class CartPage {
  static visit() { cy.visit("/index.php?route=checkout/cart&language=en-gb"); return this; }
  static rows() { return cy.get("#cart table tbody tr, .table-bordered tbody tr"); }
  static remove(idx = 0) { cy.get("button[aria-label=Remove], .fa-times").eq(idx).click(); return this; }
  static emptyMsg() { return cy.contains("cart is empty"); }
}
export default CartPage;
